### PR TITLE
Allow recalculation of effects after shrinking

### PR DESCRIPTION
**Pull request by karlb**

### CODE REVIEW

File: causing/graph.py [hunk 1]
```diff
@@ -187,10 +187,13 @@ def remove_node_keep_edges(graph, node):
     """Keep transitive connections when removing node.
 
     Removing B from A->B->C will result in A->C.
+
+    WARNING: This function only approximates the edge effects. To get accurate
+    results, you must shrink the model and recalculate the effects, instead.
```
Comment (Holger-Bartel) [inline causing/graph.py:192]: Thanks for pointing that out. We will need that even in the final version, right?

File: causing/model.py [hunk 2]
```diff
@@ -180,3 +180,18 @@ def final_ind(self):
...
         return self.xvars + self.yvars
+
+    def shrink(m: Model, remove_nodes) -> Model:  # noqa
+        """Create a model  without `remove_nodes`"""
+        yvars = []
+        equations = []
+        substitutions: list[tuple] = []
+        for yvar, eq in zip(m.yvars, m.equations):
+            if yvar in remove_nodes:
+                substitutions.insert(0, (yvar, eq))
```
Comment (Holger-Bartel) [inline causing/model.py:199]: I guess, this is an alternative to .subs?
SymPy is perfect for this job.

File: setup.py [hunk 3]
```diff
@@ -5,7 +5,7 @@
 
 setuptools.setup(
     name="causing",
-    version="2.2.0",
+    version="2.3.0",
```
Comment (Holger-Bartel) [inline setup.py:8]: Yeah, going up once again :)

File: causing/graph.py [hunk 4]
```diff
@@ -212,3 +212,45 @@ def remove_node_keep_edges(graph, node):
...
+    yhat_mean = np.mean(yhat, axis=1)
+    xdat_mean = np.mean(xdat, axis=1)
+
+    for i, approx_graph in enumerate(graphs):
+        individual_xdat = xdat[:, i : i + 1]
+        removed_nodes = set(model.graph.nodes) - set(approx_graph.nodes)
+
+        # Calc effects on shrunken model
+        individual_model = model.shrink(removed_nodes)
+        effects = individual_model.calc_effects(
```
Comment (Holger-Bartel) [inline causing/graph.py:235]: Nice and clean.

**Review by Holger-Bartel — APPROVED** [review]
Great to have graph shrinking in Causing.

File: causing/model.py (hunk 2 above)
Comment (karlb) [reply causing/model.py:199]: This is just a list of (yvar, eq) pairs. It is used in subs two lines down.

**Pull request merged**